DSCResources/MSFT_xSQLAOGroupEnsure/MSFT_xSQLAOGroupEnsure.psm1

```diff
@@ -0,0 +1,352 @@
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Present","Absent")]
+        [System.String]
+        $Ensure,
+
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $AvailabilityGroupName,
+        
+        [System.String]
+        $SQLServer = $env:COMPUTERNAME,
+
+        [ValidateNotNull()] 
+        [System.String]
+        $SQLInstanceName= "MSSQLSERVER",
+
+        [parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $SetupCredential
+    )
+
+    if(!$SQL)
+    {
+        $SQL = Connect-SQL -SQLServer $SQLServer -SQLInstanceName $SQLInstanceName
+    }
+    
+    $vConfigured = Test-TargetResource -Ensure $Ensure -AvailabilityGroupName $AvailabilityGroupName -SQLServer $SQLServer -SQLInstanceName $SQLInstanceName 
+
+    $returnValue = @{
+    Ensure = $vConfigured
+    AvailabilityGroupName = $sql.AvailabilityGroups[$AvailabilityGroupName]
+    AvailabilityGroupNameListener = $sql.AvailabilityGroups[$AvailabilityGroupName].AvailabilityGroupListeners.name
+    AvailabilityGroupNameIP = $sql.AvailabilityGroups[$AvailabilityGroupName].AvailabilityGroupListeners.availabilitygrouplisteneripaddresses.IPAddress
+    AvailabilityGroupSubMask =  $sql.AvailabilityGroups[$AvailabilityGroupName].AvailabilityGroupListeners.availabilitygrouplisteneripaddresses.SubnetMask
+    AvailabilityGroupPort =  $sql.AvailabilityGroups[$AvailabilityGroupName].AvailabilityGroupListeners.portnumber
+    AvailabilityGroupNameDatabase = $sql.AvailabilityGroups[$AvailabilityGroupName].AvailabilityDatabases.name
+    BackupDirectory = ""
+    SQLServer = $SQLServer
+    SQLInstanceName = $SQLInstanceName
+    }
+
+    $returnValue
+}
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Present","Absent")]
+        [System.String]
+        $Ensure,
+
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $AvailabilityGroupName,
+
+        [System.String]
+        $AvailabilityGroupNameListener = $AvailabilityGroupName,
+
+        [System.String[]]
+        $AvailabilityGroupNameIP,
+
+        [System.String[]]
+        $AvailabilityGroupSubMask,
+
+        [System.UInt32]
+        $AvailabilityGroupPort ="1433",
+
+        [ValidateSet("None","ReadOnly","ReadIntent")]
+        [System.String]
+        $ReadableSecondary="ReadOnly",
+
+        [ValidateSet("Primary","Secondary")]
+        [System.String]
+        $AutoBackupPrefernce ="Primary",
+
+        [System.String]
+        $SQLServer = $env:COMPUTERNAME,
+
+        [System.String]
+        $SQLInstanceName= "MSSQLSERVER",
+        
+        [parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $SetupCredential
+    )
+
+   $null = [System.Reflection.Assembly]::LoadWithPartialName('Microsoft.SqlServer.Smo')
```

Comment:
This was on my to do list however I see you have this resolved in your latest commit.  Would love to see the logic merged in.

From: Arturas Kuznecovas [mailto:notifications@github.com]
Sent: Tuesday, April 26, 2016 2:49 PM
To: PowerShell/xSQLServer <xSQLServer@noreply.github.com>
Cc: Troy Ault <troyault@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [PowerShell/xSQLServer] Always on addition (#48)


In DSCResources/MSFT_xSQLAOGroupEnsure/MSFT_xSQLAOGroupEnsure.psm1<https://github.com/PowerShell/xSQLServer/pull/48#discussion_r61142478>:

> +        [ValidateSet("Primary","Secondary")]

> +        [System.String]

> +        $AutoBackupPrefernce ="Primary",

> +

> +        [System.String]

> +        $SQLServer = $env:COMPUTERNAME,

> +

> +        [System.String]

> +        $SQLInstanceName= "MSSQLSERVER",

> +

> +        [parameter(Mandatory = $true)]

> +        [System.Management.Automation.PSCredential]

> +        $SetupCredential

> +    )

> +

> +   $null = [System.Reflection.Assembly]::LoadWithPartialName('Microsoft.SqlServer.Smo')

LoadWithPartialName is Obsolete method since .Net 2.0 as per msdn documentation.
https://msdn.microsoft.com/en-us/library/12xc5368(v=vs.110).aspx<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fmsdn.microsoft.com%2fen-us%2flibrary%2f12xc5368(v%3dvs.110).aspx&data=01%7c01%7ctroyault%40microsoft.com%7cd7458c1dc6044451758108d36e03614d%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=e0DK8L3jVNc9TZZ38DrUhX7am3%2fc324PR1T722kNXy8%3d>

This could cause issues when multiple versions of same assembly are available.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/PowerShell/xSQLServer/pull/48/files/b4994f01a2574ce0ca48e6c792d248a520d05612#r61142478>